lib/honeydew.ex

```diff
@@ -9,12 +9,41 @@ defmodule Honeydew do
   @type mod_or_mod_args :: module | {module, args :: term}
   @type queue_name :: String.t | atom | {:global, String.t | atom}
   @type supervisor_opts :: Keyword.t
+  @type async_opt :: [{:reply, true}]
+  @type task :: {atom, [arg :: term]}
+
+  @typedoc """
+  Result of a `Honeydew.Job`
+  """
+  @type result :: term
 
   #
   # Parts of this module were lovingly stolen from
   # https://github.com/elixir-lang/elixir/blob/v1.3.2/lib/elixir/lib/task.ex#L320
   #
 
+  @doc """
+  Runs a task asynchronously.
+
+  ## Examples
+
+  To run a task asynchronously.
+
+      Honeydew.async({:ping, ["127.0.0.1"]}, :my_queue)
+
+  To run a task asynchronously and wait for result.
+
+      # Without pipes
+      job = Honeydew.async({:ping, ["127.0.0.1"]}, :my_queue, reply: true)
+      Honeydew.yield(job)
+
+      # With pipes
+      result =
+        {:ping, ["127.0.0.1"]}
+        |> Honeydew.async(:my_queue, reply: true)
+        |> Honeydew.yield()
+  """
+  @spec async(task, queue_name, [async_opt]) :: Job.t | no_return
```

Comment:
why `no_return`?

Reply:
I added `no_return` because it raises an argument error when the queue doesn't exist. I guess I could add that to the docs.

Reply:
ah gotcha, please do!